ingestion_server/ingestion_server/ingest.py

```diff
@@ -288,9 +290,17 @@ def reload_upstream(
     shared_cols = _get_shared_cols(downstream_db, upstream_db, table)
     upstream_db.close()
 
-    with downstream_db.cursor() as downstream_cur:
-        # Step 2: Create FDW for the data transfer
+    with downstream_db, downstream_db.cursor() as downstream_cur:
+        # Step 2: Create the FDW extension if it does not exist
         log.info("(Re)initializing foreign data wrapper")
+        try:
+            create_ext = get_create_ext_query()
+            downstream_cur.execute(create_ext)
```

Comment:
It's interesting that this throws an exception. In postgres the `IF NOT EXISTS` clause is used to issue a notice instead of an error, and notices don't abort the current transaction so they're not _really_ exceptions. I guess it makes sense to throw on _any_ message from postgres so it can be handled in user code.

Reply:
The `IF NOT EXIST` part works fine, but in simultaneous threads, two threads can see that the extension does not exist and both try to create it, leading to an integrity error with a repetition in a primary key. So a unique violation is raised.